Output is hard to read

I ran `structslop -fix ./...` and got the following output. It does not look too bad when copy-pasted here, but it is impossible to read in the terminal.
```
/home/jacob/Git/fyne/internal/driver/glfw/canvas.go:22:15: struct has size 256 (size class 256), could be 240 (size class 240), rearrange to struct{shortcut fyne.ShortcutHandler; size fyne.Size; content fyne.CanvasObject; menu fyne.CanvasObject; context driver.WithContext; painter gl.Painter; onTypedRune func(rune); menuTree *renderCacheTree; overlays *overlayStack; onTypedKey func(*fyne.KeyEvent); onKeyDown func(*fyne.KeyEvent); menuFocusMgr *app.FocusManager; contentFocusMgr *app.FocusManager; contentTree *renderCacheTree; refreshQueue chan fyne.CanvasObject; dirtyMutex *sync.Mutex; onKeyUp func(*fyne.KeyEvent); sync.RWMutex; texScale float32; detectedScale float32; scale float32; dirty bool; padded bool} for optimal size (6.25% savings)
/home/jacob/Git/fyne/internal/driver/glfw/window.go:47:13: struct has size 400 (size class 416), could be 376 (size class 384), rearrange to struct{pending []func(); clipboard fyne.Clipboard; mousePressed fyne.CanvasObject; mouseLastClick fyne.CanvasObject; mouseOver desktop.Hoverable; mouseDraggedOffset fyne.Position; mouseDragged fyne.Draggable; title string; icon fyne.Resource; mousePos fyne.Position; mouseDragPos fyne.Position; mouseClickCount int; eventQueue chan func(); height int; width int; mainmenu *fyne.MainMenu; canvas *glCanvas; cursor *glfw.Cursor; ypos int; xpos int; mouseButton desktop.MouseButton; onCloseIntercepted func(); onClosed func(); mouseCancelFunc context.CancelFunc; viewport *glfw.Window; viewLock sync.RWMutex; eventLock sync.RWMutex; createLock sync.Once; eventWait sync.WaitGroup; mouseDragStarted bool; visible bool; centered bool; shouldExpand bool; decorate bool; fullScreen bool; fixedSize bool; master bool} for optimal size (7.69% savings)
/home/jacob/Git/fyne/dialog/fileitem.go:19:21: struct has size 136 (size class 144), could be 128 (size class 128), rearrange to struct{widget.BaseWidget; name string; location fyne.URI; picker *fileDialog; isCurrent bool; dir bool} for optimal size (11.11% savings)
/home/jacob/Git/fyne/test/testcanvas.go:30:17: struct has size 168 (size class 176), could be 160 (size class 160), rearrange to struct{fyne.ShortcutHandler; size fyne.Size; content fyne.CanvasObject; painter SoftwarePainter; hovered desktop.Hoverable; onTypedRune func(rune); onTypedKey func(*fyne.KeyEvent); overlays *internal.OverlayStack; focusMgr *app.FocusManager; propertyLock sync.RWMutex; scale float32; padded bool} for optimal size (9.09% savings)
/home/jacob/Git/fyne/internal/driver/gomobile/canvas.go:20:19: struct has size 264 (size class 288), could be 248 (size class 256), rearrange to struct{shortcut fyne.ShortcutHandler; content fyne.CanvasObject; windowHead fyne.CanvasObject; menu fyne.CanvasObject; dragging fyne.Draggable; painter gl.Painter; size fyne.Size; focused fyne.Focusable; touchLastTapped fyne.CanvasObject; onTypedKey func(event *fyne.KeyEvent); onTypedRune func(rune); touched map[int]mobile.Touchable; touchCancelFunc context.CancelFunc; touchTapCount int; lastTapDown map[int]time.Time; lastTapDownPos map[int]fyne.Position; overlays *internal.OverlayStack; refreshQueue chan fyne.CanvasObject; minSizeCache map[fyne.CanvasObject]fyne.Size; scale float32; inited bool; padded bool} for optimal size (11.11% savings)
```

Impossible to read in terminal:
![image](https://user-images.githubusercontent.com/25466657/99254304-7442f680-2812-11eb-9c5e-561f95947841.png)

Perhaps a few lines in between would make it more readable? Maybe even better formatting of it?

@Jacalz Thanks for reporting. I created PR #40 to make the output better. The output now:

```
/go/src/github.com/orijtech/structslop/testdata/struct/p.go:30:9: struct has size 24 (size class 32), could be 16 (size class 16), savings 50.00%, optimal fields order:
struct {
	y uint64
	x uint32
	z uint32
}

/go/src/github.com/orijtech/structslop/testdata/struct/p.go:36:9: struct has size 40 (size class 48), could be 24 (size class 32), savings 33.33%, optimal fields order:
struct {
	_  [0]func()
	i1 int
	i2 int
	a3 [3]bool
	b  bool
}

/go/src/github.com/orijtech/structslop/testdata/struct/p.go:59:9: struct has size 40 (size class 48), could be 32 (size class 32), savings 33.33%, optimal fields order:
struct {
	y uint64
	t *httptest.Server
	w uint64
	x uint32
	z uint32
}

/go/src/github.com/orijtech/structslop/testdata/struct/p.go:67:9: struct has size 40 (size class 48), could be 32 (size class 32), savings 33.33%, optimal fields order:
struct {
	y uint64
	t *s
	w uint64
	x uint32
	z uint32
}
```

--- a/README.md
+++ b/README.md
@@ -16,7 +16,7 @@ Without Go modules:
 
 ```sh
 $ cd $GOPATH/src/github.com/orijtech/structslop
-$ git checkout v0.0.3
+$ git checkout v0.0.4
 $ go get
 $ install ./cmd/structslop
 ```
@@ -41,10 +41,40 @@ $ structslop ./testdata/src/struct/p.go
 Sample output:
 
 ```text
-/go/src/github.com/orijtech/structslop/testdata/struct/p.go:30:9: struct has size 24 (size class 32), could be 16 (size class 16), rearrange to struct{y uint64; x uint32; z uint32} for optimal size (50.00% savings)
-/go/src/github.com/orijtech/structslop/testdata/struct/p.go:36:9: struct has size 40 (size class 48), could be 24 (size class 32), rearrange to struct{_ [0]func(); i1 int; i2 int; a3 [3]bool; b bool} for optimal size (33.33% savings)
-/go/src/github.com/orijtech/structslop/testdata/struct/p.go:59:9: struct has size 40 (size class 48), could be 32 (size class 32), rearrange to struct{y uint64; t *httptest.Server; w uint64; x uint32; z uint32} for optimal size (33.33% savings)
-/go/src/github.com/orijtech/structslop/testdata/struct/p.go:67:9: struct has size 40 (size class 48), could be 32 (size class 32), rearrange to struct{y uint64; t *s; w uint64; x uint32; z uint32} for optimal size (33.33% savings)
+/go/src/github.com/orijtech/structslop/testdata/struct/p.go:30:9: struct has size 24 (size class 32), could be 16 (size class 16), you'll save 50.00% if you rearrange it to:
+struct {
+	y uint64
+	x uint32
+	z uint32
+}
+
+/go/src/github.com/orijtech/structslop/testdata/struct/p.go:36:9: struct has size 40 (size class 48), could be 24 (size class 32), you'll save 33.33% if you rearrange it to:
+struct {
+	_  [0]func()
+	i1 int
+	i2 int
+	a3 [3]bool
+	b  bool
+}
+
+/go/src/github.com/orijtech/structslop/testdata/struct/p.go:59:9: struct has size 40 (size class 48), could be 32 (size class 32), you'll save 33.33% if you rearrange it to:
+struct {
+	y uint64
+	t *httptest.Server
+	w uint64
+	x uint32
+	z uint32
+}
+
+/go/src/github.com/orijtech/structslop/testdata/struct/p.go:67:9: struct has size 40 (size class 48), could be 32 (size class 32), you'll save 33.33% if you rearrange it to:
+struct {
+	y uint64
+	t *s
+	w uint64
+	x uint32
+	z uint32
+}
+
 ```
 
 Example, for the first report above, the output meaning:
@@ -85,7 +115,13 @@ However, you can still get this information when you want, using `-verbose` flag
 $ structslop -verbose ./testdata/src/verbose/p.go
 /go/src/github.com/orijtech/structslop/testdata/src/verbose/p.go:17:8: struct has size 0 (size class 0)
 /go/src/github.com/orijtech/structslop/testdata/src/verbose/p.go:19:9: struct has size 1 (size class 8)
-/go/src/github.com/orijtech/structslop/testdata/src/verbose/p.go:23:9: struct has size 32 (size class 32), could be 24 (size class 32), rearrange to struct{y uint64; z *s; x uint32; t uint32} for optimal size
+/go/src/github.com/orijtech/structslop/testdata/src/verbose/p.go:23:9: struct has size 32 (size class 32), could be 24 (size class 32), optimal fields order:
+struct {
+	y uint64
+	z *s
+	x uint32
+	t uint32
+}
 ```
  
 ## Development

--- a/fields_preserve_test.go
+++ b/fields_preserve_test.go
@@ -54,7 +54,7 @@ type s struct {
 		if atyp, ok := n.(*ast.StructType); ok {
 			if tv, ok := info.Types[atyp]; ok {
 				styp := tv.Type.(*types.Struct)
-				optStruct := optimalStructArrangement(sizes, styp)
+				optStruct := optimalStructArrangement(sizes, mapFieldIdx(styp))
 				if optStruct.Field(0) != styp.Field(0) {
 					t.Errorf("%v field order changed", styp.Field(0))
 				}

--- a/structslop.go
+++ b/structslop.go
@@ -15,9 +15,13 @@
 package structslop
 
 import (
+	"bytes"
 	"fmt"
 	"go/ast"
 	"go/build"
+	"go/format"
+	"go/parser"
+	"go/token"
 	"go/types"
 	"sort"
 	"strings"
@@ -80,33 +84,52 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			return
 		}
 
+		var buf bytes.Buffer
+		if err := format.Node(&buf, pass.Fset, atyp); err != nil {
+			return
+		}
+		expr, err := parser.ParseExpr(buf.String())
+		if err != nil {
+			return
+		}
+		natyp := expr.(*ast.StructType)
+		fl := make([]*ast.Field, styp.NumFields())
+		for i, idx := range r.optIdx {
+			fl[i] = natyp.Fields.List[idx]
+		}
+		natyp.Fields.List = fl
+
+		buf.Reset()
+		if err := format.Node(&buf, token.NewFileSet(), natyp); err != nil {
+			return
+		}
+
 		var msg string
 		switch {
 		case r.oldGcSize == r.newGcSize:
 			msg = fmt.Sprintf("struct has size %d (size class %d)", r.oldGcSize, r.oldRuntimeSize)
 		case r.oldGcSize != r.newGcSize:
-			curPkgPath := pass.Pkg.Path()
-			optStyp := formatStruct(r.suggestedStruct, curPkgPath)
 			msg = fmt.Sprintf(
-				"struct has size %d (size class %d), could be %d (size class %d), rearrange to %s for optimal size",
+				"struct has size %d (size class %d), could be %d (size class %d), optimal fields order:\n%s\n",
 				r.oldGcSize,
 				r.oldRuntimeSize,
 				r.newGcSize,
 				r.newRuntimeSize,
-				optStyp,
+				buf.String(),
 			)
 			if r.sloppy() {
 				msg = fmt.Sprintf(
-					"struct has size %d (size class %d), could be %d (size class %d), rearrange to %s for optimal size (%.2f%% savings)",
+					"struct has size %d (size class %d), could be %d (size class %d), you'll save %.2f%% if you rearrange it to:\n%s\n",
 					r.oldGcSize,
 					r.oldRuntimeSize,
 					r.newGcSize,
 					r.newRuntimeSize,
-					optStyp,
 					r.savings(),
+					buf.String(),
 				)
 			}
 		}
+
 		pass.Report(analysis.Diagnostic{
 			Pos:            n.Pos(),
 			End:            n.End(),
@@ -118,11 +141,12 @@ func run(pass *analysis.Pass) (interface{}, error) {
 }
 
 type result struct {
-	oldGcSize       int64
-	newGcSize       int64
-	oldRuntimeSize  int64
-	newRuntimeSize  int64
-	suggestedStruct *types.Struct
+	oldGcSize      int64
+	newGcSize      int64
+	oldRuntimeSize int64
+	newRuntimeSize int64
+	optStruct      *types.Struct
+	optIdx         []int
 }
 
 func (r result) sloppy() bool {
@@ -133,23 +157,36 @@ func (r result) savings() float64 {
 	return float64(r.oldRuntimeSize-r.newRuntimeSize) / float64(r.oldRuntimeSize) * 100
 }
 
+func mapFieldIdx(s *types.Struct) map[*types.Var]int {
+	m := make(map[*types.Var]int, s.NumFields())
+	for i := 0; i < s.NumFields(); i++ {
+		m[s.Field(i)] = i
+	}
+	return m
+}
+
 func checkSloppy(pass *analysis.Pass, origStruct *types.Struct) result {
-	optStruct := optimalStructArrangement(pass.TypesSizes, origStruct)
+	m := mapFieldIdx(origStruct)
+	optStruct := optimalStructArrangement(pass.TypesSizes, m)
+	idx := make([]int, optStruct.NumFields())
+	for i := range idx {
+		idx[i] = m[optStruct.Field(i)]
+	}
 	r := result{
-		oldGcSize:       pass.TypesSizes.Sizeof(origStruct),
-		newGcSize:       pass.TypesSizes.Sizeof(optStruct),
-		suggestedStruct: optStruct,
+		oldGcSize: pass.TypesSizes.Sizeof(origStruct),
+		newGcSize: pass.TypesSizes.Sizeof(optStruct),
+		optStruct: optStruct,
+		optIdx:    idx,
 	}
 	r.oldRuntimeSize = int64(roundUpSize(uintptr(r.oldGcSize)))
 	r.newRuntimeSize = int64(roundUpSize(uintptr(r.newGcSize)))
 	return r
 }
 
-func optimalStructArrangement(sizes types.Sizes, s *types.Struct) *types.Struct {
-	nf := s.NumFields()
-	fields := make([]*types.Var, nf)
-	for i := 0; i < nf; i++ {
-		fields[i] = s.Field(i)
+func optimalStructArrangement(sizes types.Sizes, m map[*types.Var]int) *types.Struct {
+	fields := make([]*types.Var, len(m))
+	for v, i := range m {
+		fields[i] = v
 	}
 
 	sort.Slice(fields, func(i, j int) bool {
@@ -176,14 +213,4 @@ func optimalStructArrangement(sizes types.Sizes, s *types.Struct) *types.Struct 
 	})
 
 	return types.NewStruct(fields, nil)
-}
-
-func formatStruct(styp *types.Struct, curPkgPath string) string {
-	qualifier := func(p *types.Package) string {
-		if p.Path() == curPkgPath {
-			return ""
-		}
-		return p.Name()
-	}
-	return types.TypeString(styp, qualifier)
 }

--- a/testdata/src/include-test-files/p.go
+++ b/testdata/src/include-test-files/p.go
@@ -14,7 +14,7 @@
 
 package p
 
-type s struct { // want `struct has size 24 \(size class 32\), could be 16 \(size class 16\), rearrange to struct{y uint64; x uint32; z uint32} for optimal size \(50.00% savings\)`
+type s struct { // want `struct has size 24 \(size class 32\), could be 16 \(size class 16\), you'll save 50.00% if you rearrange it to:\nstruct {\n\ty uint64\n\tx uint32\n\tz uint32\n}`
 	x uint32
 	y uint64
 	z uint32

--- a/testdata/src/struct/p.go
+++ b/testdata/src/struct/p.go
@@ -27,13 +27,13 @@ type s2 struct {
 	j int
 }
 
-type s3 struct { // want `struct has size 24 \(size class 32\), could be 16 \(size class 16\), rearrange to struct{y uint64; x uint32; z uint32} for optimal size \(50.00% savings\)`
+type s3 struct { // want `struct has size 24 \(size class 32\), could be 16 \(size class 16\), you'll save 50.00% if you rearrange it to:\nstruct {\n\ty uint64\n\tx uint32\n\tz uint32\n}`
 	x uint32
 	y uint64
 	z uint32
 }
 
-type s4 struct { // want `struct has size 40 \(size class 48\), could be 24 \(size class 32\), rearrange to struct{_ \[0\]func\(\); i1 int; i2 int; a3 \[3\]bool; b bool} for optimal size \(33.33% savings\)`
+type s4 struct { // want `struct has size 40 \(size class 48\), could be 24 \(size class 32\), you'll save 33.33% if you rearrange it to:\nstruct {\n\t_  \[0\]func\(\)\n\ti1 int\n\ti2 int\n\ta3 \[3\]bool\n\tb  bool\n}`
 	b  bool
 	i1 int
 	i2 int
@@ -56,7 +56,7 @@ type s6 struct { // should be good, see #16
 	index uintptr
 }
 
-type s7 struct { // want `struct has size 40 \(size class 48\), could be 32 \(size class 32\), rearrange to struct{y uint64; t \*httptest.Server; w uint64; x uint32; z uint32} for optimal size \(33.33% savings\)`
+type s7 struct { // want `struct has size 40 \(size class 48\), could be 32 \(size class 32\), you'll save 33.33% if you rearrange it to:\nstruct {\n\ty uint64\n\tt \*httptest.Server\n\tw uint64\n\tx uint32\n\tz uint32\n}`
 	x uint32
 	y uint64
 	t *httptest.Server
@@ -64,7 +64,7 @@ type s7 struct { // want `struct has size 40 \(size class 48\), could be 32 \(si
 	w uint64
 }
 
-type s8 struct { // want `struct has size 40 \(size class 48\), could be 32 \(size class 32\), rearrange to struct{y uint64; t \*s; w uint64; x uint32; z uint32} for optimal size \(33.33% savings\)`
+type s8 struct { // want `struct has size 40 \(size class 48\), could be 32 \(size class 32\), you'll save 33.33% if you rearrange it to:\nstruct {\n\ty uint64\n\tt \*s\n\tw uint64\n\tx uint32\n\tz uint32\n}`
 	x uint32
 	y uint64
 	t *s

--- a/testdata/src/verbose/p.go
+++ b/testdata/src/verbose/p.go
@@ -20,7 +20,7 @@ type s1 struct { // want `struct has size 1 \(size class 8\)`
 	b bool
 }
 
-type s3 struct { // want `struct has size 32 \(size class 32\), could be 24 \(size class 32\), rearrange to struct{y uint64; z \*s; x uint32; t uint32} for optimal size`
+type s3 struct { // want `struct has size 32 \(size class 32\), could be 24 \(size class 32\), optimal fields order:\nstruct {\n\ty uint64\n\tz \*s\n\tx uint32\n\tt uint32\n}`
 	x uint32
 	y uint64
 	z *s

